.template/lib/main.dart

```diff
@@ -65,10 +84,55 @@ class MyHomePage extends StatelessWidget {
             Text(
               FlutterConfig.get('SECRET'),
               style: const TextStyle(color: Colors.black, fontSize: 24),
-            )
+            ),
+            const SizedBox(height: 24),
+            ElevatedButton(
+              onPressed: () => context.go('/page2'),
+              child: const Text("Navigate to Page 2"),
+            ),
           ],
         ),
       ),
     );
   }
 }
+
+class Page3Screen extends StatelessWidget {
+  const Page3Screen({
+    Key? key,
+  }) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: const Text("Page 3"),
+      ),
+      body: Center(
+          child: ElevatedButton(
+        onPressed: () => context.pop(),
+        child: const Text("Back"),
+      )),
+    );
+  }
+}
+
+class Page2Screen extends StatelessWidget {
+  const Page2Screen({
+    Key? key,
+  }) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: const Text("Page 2"),
+      ),
+      body: Center(
+          child: ElevatedButton(
+        onPressed: () => context.push('/page3'),
```

Comment:
What do you think about adding an example about pushing with parameters? 🤔

Reply:
@markgravity I don't want to add too many examples into our template app, demonstrating the `go` and `push` functions are enough IMO. If we want to deep dive into `go_router` features, we can create an app in [flutter-showcases](https://github.com/nimblehq/flutter-showcases) repo for that purpose I think 😄